torch_frame/nn/encoder/stype_encoder.py

```diff
@@ -19,10 +20,32 @@
 from torch_frame.data.multi_nested_tensor import MultiNestedTensor
 from torch_frame.data.stats import StatType
 from torch_frame.nn.base import Module
+from torch_frame.nn.encoding import CyclicEncoding, PositionalEncoding
 from torch_frame.typing import TensorData
 
 from ..utils.init import attenuated_kaiming_uniform_
 
+NUM_MONTHS_PER_YEAR = 12
+'''
+TODO: We might want to revisit the normalization
+constant for NUM_DAYS_PER_MONTH as some months only
+have 28-30 days.
+'''
+NUM_DAYS_PER_MONTH = 31
+NUM_DAYS_PER_WEEK = 7
+NUM_HOURS_PER_DAY = 24
+NUM_MINUTES_PER_HOUR = 60
+NUM_SECONDS_PER_MINUTE = 60
+TIME_TO_INDEX = {
+    'YEAR': 0,
+    'MONTH': 1,
+    'DAY': 2,
+    'DAYOFWEEK': 3,
+    'HOUR': 4,
+    'MINUTE': 5,
+    'SECOND': 6
+}
```

Comment:
Can we derive this from `TimestampTensorMapper`? The idea is to keep this ordering in a single place.